### PR TITLE
【KernelGen】Add poisson operator

### PR DESCRIPTION
**Pull request by Schopenhauer-loves-Hegel**
### PR Category
Operator

### Type of Change
New Feature

### Description
Add `poisson` operator implementation with Triton kernel.

- Implementation mode: `N/A`
- Accuracy test: 12/12 passed

### Issue
N/A

### Progress

- [ ] Change is properly reviewed (1 reviewer required, 2 recommended).
- [ ] Change is responded to an issue.
- [x] Change is fully covered by a UT.

### Performance
**torch.bfloat16**

| Shape | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|-------|-------------------:|-------------------:|--------:|
| torch.Size([1073741824]) | 54.8450 | 342.9034 | 0.160 |
| torch.Size([64, 64]) | 0.0115 | 0.0152 | 0.758 |
| torch.Size([4096, 4096]) | 0.8674 | 5.3705 | 0.162 |
| torch.Size([64, 512, 512]) | 0.8681 | 5.3700 | 0.162 |
| torch.Size([1024, 1024, 1024]) | 54.8514 | 342.8860 | 0.160 |
| torch.Size([268435456]) | 13.7180 | 85.7287 | 0.160 |
| torch.Size([10000, 1]) | 0.0122 | 0.0172 | 0.711 |
| torch.Size([10000, 256]) | 0.1417 | 0.8301 | 0.171 |
| torch.Size([10000, 65536]) | 33.5239 | 209.2859 | 0.160 |
| torch.Size([100, 1, 100]) | 0.0123 | 0.0172 | 0.717 |
| torch.Size([100, 256, 100]) | 0.1413 | 0.8301 | 0.170 |
| torch.Size([100, 65536, 100]) | 33.5589 | 209.2817 | 0.160 |

**torch.float16**

| Shape | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|-------|-------------------:|-------------------:|--------:|
| torch.Size([1073741824]) | 54.9470 | 342.3941 | 0.160 |
| torch.Size([64, 64]) | 0.0123 | 0.0147 | 0.836 |
| torch.Size([4096, 4096]) | 0.8686 | 5.3617 | 0.162 |
| torch.Size([64, 512, 512]) | 0.8687 | 5.3613 | 0.162 |
| torch.Size([1024, 1024, 1024]) | 54.9582 | 342.3581 | 0.161 |
| torch.Size([268435456]) | 13.7483 | 85.6007 | 0.161 |
| torch.Size([10000, 1]) | 0.0123 | 0.0175 | 0.700 |
| torch.Size([10000, 256]) | 0.1412 | 0.8288 | 0.170 |
| torch.Size([10000, 65536]) | 33.5449 | 208.9489 | 0.161 |
| torch.Size([100, 1, 100]) | 0.0116 | 0.0172 | 0.674 |
| torch.Size([100, 256, 100]) | 0.1412 | 0.8290 | 0.170 |
| torch.Size([100, 65536, 100]) | 33.5399 | 208.9189 | 0.161 |

**torch.float32**

| Shape | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|-------|-------------------:|-------------------:|--------:|
| torch.Size([1073741824]) | 55.2557 | 357.3297 | 0.155 |
| torch.Size([64, 64]) | 0.0115 | 0.0183 | 0.629 |
| torch.Size([4096, 4096]) | 0.8740 | 5.5972 | 0.156 |
| torch.Size([64, 512, 512]) | 0.8735 | 5.5975 | 0.156 |
| torch.Size([1024, 1024, 1024]) | 55.2631 | 357.3276 | 0.155 |
| torch.Size([268435456]) | 13.8201 | 89.3406 | 0.155 |
| torch.Size([10000, 1]) | 0.0124 | 0.0203 | 0.610 |
| torch.Size([10000, 256]) | 0.1421 | 0.8665 | 0.164 |
| torch.Size([10000, 65536]) | 33.7218 | 218.0784 | 0.155 |
| torch.Size([100, 1, 100]) | 0.0124 | 0.0203 | 0.611 |
| torch.Size([100, 256, 100]) | 0.1421 | 0.8675 | 0.164 |
| torch.Size([100, 65536, 100]) | 33.7381 | 218.1550 | 0.155 |

**Overall: median speedup = 0.162x, mean speedup = 0.294x** (36 data points)

---
_Generated by auto_gen tool with Claude Code_

### CODE REVIEW

**Comment by factnn** [comment]
## Benchmark Results (poisson)

Tested on local GPU environment.

**torch.float16**

| Size | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|------|-------------------:|-------------------:|--------:|
| [1073741824] | 79.120 | 523.271 | 0.151 |
| [64, 64] | 0.012 | 0.038 | 0.317 |
| [4096, 4096] | 0.869 | 8.493 | 0.102 |
| [64, 512, 512] | 0.871 | 8.490 | 0.103 |
| [1024, 1024, 1024] | 141.839 | 526.282 | 0.270 |
| [268435456] | 19.402 | 135.833 | 0.143 |
| [10000, 1] | 0.011 | 0.037 | 0.311 |
| [10000, 256] | 0.142 | 0.892 | 0.159 |
| [10000, 65536] | 93.638 | 319.054 | 0.293 |
| [100, 1, 100] | 0.011 | 0.038 | 0.301 |
| [100, 256, 100] | 0.142 | 0.892 | 0.159 |
| [100, 65536, 100] | 46.756 | 321.612 | 0.145 |

**torch.float32**

| Size | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|------|-------------------:|-------------------:|--------:|
| [1073741824] | 78.544 | 515.947 | 0.152 |
| [64, 64] | 0.011 | 0.038 | 0.297 |
| [4096, 4096] | 0.874 | 8.468 | 0.103 |
| [64, 512, 512] | 0.874 | 8.789 | 0.099 |
| [1024, 1024, 1024] | 77.225 | 524.395 | 0.147 |
| [268435456] | 19.291 | 132.057 | 0.146 |
| [10000, 1] | 0.011 | 0.037 | 0.311 |
| [10000, 256] | 0.142 | 0.890 | 0.160 |
| [10000, 65536] | 47.357 | 589.461 | 0.080 |
| [100, 1, 100] | 0.012 | 0.038 | 0.323 |
| [100, 256, 100] | 0.142 | 0.889 | 0.160 |
| [100, 65536, 100] | 47.199 | 322.456 | 0.146 |

**torch.bfloat16**

| Size | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|------|-------------------:|-------------------:|--------:|
| [1073741824] | 77.766 | 664.307 | 0.117 |
| [64, 64] | 0.012 | 0.051 | 0.237 |
| [4096, 4096] | 0.869 | 8.506 | 0.102 |
| [64, 512, 512] | 0.869 | 7.877 | 0.110 |
| [1024, 1024, 1024] | 75.688 | 526.312 | 0.144 |
| [268435456] | 19.129 | 131.743 | 0.145 |
| [10000, 1] | 0.011 | 0.051 | 0.226 |
| [10000, 256] | 0.142 | 0.909 | 0.156 |
| [10000, 65536] | 47.030 | 591.651 | 0.079 |
| [100, 1, 100] | 0.011 | 0.051 | 0.226 |
| [100, 256, 100] | 0.141 | 0.908 | 0.155 |
| [100, 65536, 100] | 57.055 | 562.997 | 0.101 |

All sizes show Gems speedup < 1. The Triton kernel is significantly slower than the native PyTorch implementation.

**Comment by tengqm** [comment]
performance is pretty bad, any thoughts on improving it?
Conflicts must be resolved.

File: benchmark/test_poisson_perf.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import pytest
+import torch
+
+from benchmark.performance_utils import GenericBenchmark2DOnly
```
Comment (tengqm) [inline benchmark/test_poisson_perf.py]: Rebase to master.

File: tests/test_poisson.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    var = torch.var(ref_out.to(torch.float32))
+
+    assert torch.abs(mean - lam) < 1.0
+    assert torch.abs(var - lam) < 5.0
+    assert (res_out >= 0).all()
+
+
+@pytest.mark.poisson
+@pytest.mark.parametrize("dtype", FLOAT_DTYPES)
+def test_accuracy_poisson_zero_rate(dtype):
```
Comment (tengqm) [inline tests/test_poisson.py]: Remove `_accuracy` from this and the above function names. It is redundant.

**Comment by factnn** [comment]
## UT Results

Branch rebased to upstream/master, test function names updated (removed `accuracy_` prefix), benchmark updated to new format.

```
python -m pytest tests/test_poisson.py -x -q
12 passed
```

File: src/flag_gems/ops/poisson.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    Uses inverse transform: -log(U) / lam for exponential samples.
+    """
+    # L = exp(-lambda)
+    L = tl.exp(-lam)
+    k = (lam * 0).to(tl.int32)  # Initialize counter to 0
+    p = lam * 0.0 + 1.0  # Initialize p to 1.0
+
+    # We need to iterate. Each iteration we multiply p by a uniform random.
+    # Continue while p > L.
+    for _ in range(MAX_ITERS):
```
Comment (w1120029931-bit) [inline src/flag_gems/ops/poisson.py:32]: The current performance is too poor. Each element simultaneously performs Knuth 64 rounds + Box-Muller. Is it possible to split the algorithm path to improve performance.

**Comment by factnn** [comment]
## Performance Benchmark Results

```
Operator: poisson  Performance Test (dtype=torch.float16, mode=kernel,level=comprehensive)
Status       Torch Latency (ms)    Gems Latency (ms)         Gems Speedup          Size Detail
-----------------------------------------------------------------------------------------------
SUCCESS               0.012064            0.017408               0.693          [torch.Size([64, 64])]
SUCCESS               0.013472            0.035200               0.383          [torch.Size([256, 256])]
SUCCESS               0.032320            0.343808               0.094          [torch.Size([1024, 1024])]
SUCCESS               0.377952            5.354336               0.071          [torch.Size([4096, 4096])]

Operator: poisson  Performance Test (dtype=torch.float32, mode=kernel,level=comprehensive)
Status       Torch Latency (ms)    Gems Latency (ms)         Gems Speedup          Size Detail
-----------------------------------------------------------------------------------------------
SUCCESS               0.010016            0.017312               0.579          [torch.Size([64, 64])]
SUCCESS               0.010048            0.037824               0.266          [torch.Size([256, 256])]
SUCCESS               0.033600            0.357792               0.094          [torch.Size([1024, 1024])]
SUCCESS               0.384880            5.551584               0.069          [torch.Size([4096, 4096])]

Operator: poisson  Performance Test (dtype=torch.bfloat16, mode=kernel,level=comprehensive)
Status       Torch Latency (ms)    Gems Latency (ms)         Gems Speedup          Size Detail
-----------------------------------------------------------------------------------------------
SUCCESS               0.010016            0.013920               0.720          [torch.Size([64, 64])]
SUCCESS               0.011424            0.035200               0.325          [torch.Size([256, 256])]
SUCCESS               0.033248            0.346784               0.096          [torch.Size([1024, 1024])]
SUCCESS               0.378048            5.383296               0.070          [torch.Size([4096, 4096])]
```

**Comment by factnn** [comment]
Both suggestions have been addressed: the benchmark file has been renamed(removing _accuracy), and rebasing to master.

**Review by tengqm — APPROVED** [review]

**Review by w1120029931-bit — APPROVED** [review]
